src/main/resources/ath-container/Dockerfile

```diff
@@ -37,6 +37,13 @@ RUN install -m 0755 -d /etc/apt/keyrings \
   && apt-get clean \
   && rm -rf /var/lib/apt/lists/*
 
+# Despite the docker SUID hack below, test-containers accesses /var/run/docker.sock directly and so we can not rely on the SUID hack.
+# Rather take the docker user group as an arg and make the ath-user a member of that group
+# we retain the suid workaround as this method requires a local build of the container
+# we need to do this before we install docker so that any files have the correct permission
+ARG dockergid=1002
+RUN groupadd docker -g $dockergid
```

Comment:
Replaying the build with these `Jenkinsfile` changes, it fails:

```
12:49:01   > [ 4/18] RUN groupadd docker -g 999:
12:49:01  0.172 groupadd: GID '999' already exists
```

Reply:
Well that sucks...

Reply:
For debugging, we have the following in the ci.jio. environment:

```shell
[2024-09-04T19:45:51.737Z] ++ id -u
[2024-09-04T19:45:51.738Z] ++ id -g
[2024-09-04T19:45:51.738Z] ++ getent group docker
[2024-09-04T19:45:51.739Z] ++ cut -d: -f3
[2024-09-04T19:45:51.739Z] + docker build -t jenkins/ath --build-arg uid=1001 --build-arg gid=1001 --build-arg dockergid=999 ./src/main/resources/ath-container/
```

For the record, we are using the Docker CE package from <https://docs.docker.com/engine/install/ubuntu/> as per https://github.com/jenkins-infra/packer-images/blob/1.94.0/provisioning/ubuntu-provision.sh#L187-L205 to install it. I believe we depends on their setup.
But it could be worth defining a value for the UID/GID for the `docker` user/group and set it up in the install: as far as I know ATH is the only "weird" usage of Docker so I don't mind setting up such a contract with the infra image template. It would be easy to check as we have an [automated test harness](https://github.com/jenkins-infra/packer-images/blob/1.94.0/tests/goss-linux.yaml) to describe such kind of requirement and avoid loosing it in hte future.